packages/tools/src/tools/base/BaseTool.ts

```diff
@@ -83,7 +84,38 @@ abstract class BaseTool implements IBaseTool {
     operationData: unknown
   ): any {
     const { strategies, activeStrategy } = this.configuration;
-    return strategies[activeStrategy].call(this, enabledElement, operationData);
+    return strategies[activeStrategy]?.call(
+      this,
+      enabledElement,
+      operationData
+    );
+  }
+
+  /**
+   * Applies the active strategy, with a given event type being applied.
+   * The event type function is found by indexing it on the active strategy
+   * function.
+   *
+   * @param enabledElement - The element that is being operated on.
+   * @param operationData - The data that needs to be passed to the strategy.
+   * @returns The result of the strategy.
+   */
+  public applyActiveStrategyCallback(
+    enabledElement: Types.IEnabledElement,
+    operationData: unknown,
+    callbackType:
+      | 'initDown'
+      | 'completeUp'
+      | 'rejectPreview'
+      | 'acceptPreview'
+      | 'preview'
```

Comment:
Having an accept/reject preview here in the base tool is quite unusual. Maybe these are only specific to the Brush Tool?

Reply:
Previews aren't specific to the brush tool - we happen to only use it there, but in general it is a useful concept.

Reply:
what would be the preview for annotation?